From 2b3ed2d0a670763827a3872418d59995fdf398fa Mon Sep 17 00:00:00 2001
From: Joey Riches <josephriches@gmail.com>
Date: Mon, 10 Jan 2022 14:08:16 +0000
Subject: [PATCH 1/1] Allow errors to pass through on lstat walk

abi-wizard is sometimes choking on random files in %installroot%
and I don't know why.
---
 abi/report.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/abi/report.go b/abi/report.go
index f88955d..78f25a8 100644
--- a/abi/report.go
+++ b/abi/report.go
@@ -122,7 +122,7 @@ func (r Report) Save(path string) error {
 func (r Report) Add(root, path string) error {
 	info, err := os.Lstat(path)
 	if err != nil {
-		return err
+		return nil
 	}
 	return r.walkPivot(root, path, info)
 }
@@ -166,7 +166,7 @@ func (r Report) walkSym(root, path string) error {
 	} // implicit else: abs path outside root
 	info, err := os.Lstat(link)
 	if err != nil {
-		return err
+		return nil
 	}
 	return r.walkPivot(root, link, info)
 }
-- 
2.34.0

